src/main/scala/scala/compat/java8/collectionImpl/Accumulator.scala

```diff
@@ -0,0 +1,344 @@
+      if (current.length - index >= n) {
+        System.arraycopy(that.history(h), 0, current, index, n)
+        prev = that.cumulative(h)
+        index += n
+        h += 1
+      }
+      else more = false
+    }
+    if (h >= that.hIndex && current.length - index >= that.index) {
+      if (that.index > 0) System.arraycopy(that.current, 0, current, index, that.index)
+      index += that.index
+    }
+    else {
+      val slots = (if (index > 0) 1 else 0) + that.hIndex - h
+      if (hIndex + slots > history.length) {
+        val n = math.max(4, 1 << (32 - java.lang.Integer.numberOfLeadingZeros(1 + hIndex + slots)))
+        history = java.util.Arrays.copyOf(history, n)
+        cumul = java.util.Arrays.copyOf(cumul, n)
+      }
+      var pv = (if (hIndex > 0) cumulative(hIndex-1) else 0L)
+      if (index > 0) {
+        pv += index
+        cumul(hIndex) = pv
+        history(hIndex) = (if (index < (current.length >>> 3) && current.length > 32) java.util.Arrays.copyOf(current, index) else current)
+        hIndex += 1
+      }
+      while (h < that.hIndex) {
+        pv += that.cumulative(h) - prev
+        prev = that.cumulative(h)
+        cumul(hIndex) = pv
+        history(hIndex) = that.history(h)
+        h += 1
+        hIndex += 1
+      }
+      index = that.index
+      current = that.current
+    }
+    totalSize += that.totalSize
+    that.clear
+  }
+  
+  override def clear(): Unit = {
+    super.clear()
+    current = Accumulator.emptyAnyRefArray
+    history = Accumulator.emptyAnyRefArrayArray
+    cumul  = Accumulator.emptyLongArray
+  }
+  
+  /** Retrieves the `ix`th element. */
+  final def apply(ix: Long): A = {
+    if (totalSize - ix <= index || hIndex == 0) current((ix - (totalSize - index)).toInt).asInstanceOf[A]
+    else {
+      val w = seekSlot(ix)
+      history((w >>> 32).toInt)((w & 0xFFFFFFFFL).toInt).asInstanceOf[A]
+    }
+  }
+  
+  /** Retrieves the `ix`th element, using an `Int` index. */
+  final def apply(i: Int): A = apply(i.toLong)
+  
+  /** Returns a `Stepper` over the contents of this `Accumulator`*/
+  final def stepper: AnyStepper[A] = new AccumulatorStepper[A](this)
+  
+  /** Returns an `Iterator` over the contents of this `Accumulator`. */
+  final def iterator = stepper.iterator
+
+  /** Returns a `java.util.Spliterator` over the contents of this `Accumulator`*/
+  final def spliterator: java.util.Spliterator[A] = stepper.spliterator
+  
+  /** Produces a sequential Java 8 Stream over the elements of this `Accumulator`*/
+  final def seqStream: java.util.stream.Stream[A] = java.util.stream.StreamSupport.stream(spliterator, false)
+  
+  /** Produces a parallel Java 8 Stream over the elements of this `Accumulator`*/
+  final def parStream: java.util.stream.Stream[A] = java.util.stream.StreamSupport.stream(spliterator, true)
+  
+  /** Copies the elements in this `Accumulator` into an `Array` */
+  final def toArray(implicit tag: reflect.ClassTag[A]) = {
```

Comment:
You mean copying from an `Accumulator[Int]` which contains buffers of type `Array[java.lang.Integer]` to an `Array[Int]`? Could this case be prevented by using the right buffer type in the first place?